implementations/rust/ockam/ockam_entity/src/credential/workers/holder.rs

```diff
@@ -0,0 +1,128 @@
+use crate::{
+    check_message_origin, Credential, CredentialAttribute, CredentialFragment1,
+    CredentialProtocolMessage, CredentialSchema, EntityCredential, EntityError, Holder, Identity,
+    Profile, ProfileIdentifier,
+};
+use async_trait::async_trait;
+use ockam_core::lib::convert::TryInto;
+use ockam_core::{Address, Result, Route, Routed, Worker};
+use ockam_node::Context;
+use rand::random;
+
+enum State {
+    One,
+    Two([u8; 96], CredentialFragment1),
+    Done,
+}
+
+pub struct HolderWorker {
+    state: State,
+    profile: Profile,
+    issuer_id: ProfileIdentifier,
+    issuer_route: Route,
+    schema: CredentialSchema,
+    values: Vec<CredentialAttribute>,
+    callback_address: Address,
+}
+
+impl HolderWorker {
+    pub fn new(
+        profile: Profile,
+        issuer_id: ProfileIdentifier,
+        issuer_route: Route,
+        schema: CredentialSchema,
+        values: Vec<CredentialAttribute>,
+        callback_address: Address,
+    ) -> Self {
+        Self {
+            profile,
+            issuer_id,
+            state: State::One,
+            issuer_route,
+            schema,
+            values,
+            callback_address,
+        }
+    }
+}
+
+#[async_trait]
+impl Worker for HolderWorker {
+    type Context = Context;
+    type Message = CredentialProtocolMessage;
+
+    async fn initialize(&mut self, ctx: &mut Self::Context) -> Result<()> {
+        ctx.send(
+            self.issuer_route.clone(),
+            CredentialProtocolMessage::IssueOfferRequest(self.schema.id.clone()),
+        )
+        .await?;
+
+        Ok(())
```

Comment:
```suggestion
        .await
```

Reply:
I actually prefer returning Ok(()), because it's easier to modify such functions